### PR TITLE
chore: rename cdk8s-plus to cdk8s-plus-17

### PR DESCRIPTION
**Pull request by iliapolo**
In preparation for supporting multiple kubernetes api specs, we are beginning to vend a module for each API spec. Starting with `1.17.0`.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk8s-cli/src/cli/cmds/init.ts [hunk 1]
```diff
@@ -44,11 +44,11 @@ async function determineDeps(version: string, dist?: string): Promise<Deps> {
     const ret = {
       npm_cdk8s: path.resolve(dist, 'js', `cdk8s@${version}.jsii.tgz`),
       npm_cdk8s_cli: path.resolve(dist, 'js', `cdk8s-cli-v${version}.tgz`), // yarn pack adds a "v" before the version
-      npm_cdk8s_plus: path.resolve(dist, 'js', `cdk8s-plus@${version}.jsii.tgz`),
+      npm_cdk8s_plus: path.resolve(dist, 'js', `cdk8s-plus-17@${version}.jsii.tgz`),
```
Comment (eladb) [inline packages/cdk8s-cli/src/cli/cmds/init.ts:47]: Curious what we'll do with this

**Review by eladb — DISMISSED** [review]

File: packages/cdk8s-cli/src/cli/cmds/init.ts (hunk 1 above)
Comment (iliapolo) [reply packages/cdk8s-cli/src/cli/cmds/init.ts:47]: Maybe just a dynamic search for `cdk8s-plus*`

Comment (eladb) [reply packages/cdk8s-cli/src/cli/cmds/init.ts:47]: Let's figure out later

File: docs/plus/index.md [hunk 2]
```diff
@@ -7,16 +7,27 @@ Built on top of the auto-generated building blocks provided by CDK8s, this
 library includes a hand crafted *construct* for each native kubernetes object,
 exposing richer API's with reduced complexity.
 
+**cdk8s+** is vended as a separate library for each kubernetes spec version. The documentation presented here represents version [1.17.0](https://github.com/kubernetes/kubernetes/tree/v1.17.0/api/openapi-spec)
+and is vended as the `cdk8s-plus-17` library.
+
+### Naming conventions
+
+- Stable resources are represented by a *construct* of the same name **base** name. For example, the `io.k8s.api.core.v1.Pod` resource maps to the `Pod` *construct*.
```
Comment (eladb) [inline docs/plus/index.md:15]:
```suggestion
- Stable resources are represented by a *construct* of the same name **kind** name. For example, the `io.k8s.api.core.v1.Pod` resource maps to the `Pod` *construct*.
```

File: docs/plus/index.md [hunk 3]
```diff
@@ -7,16 +7,27 @@ Built on top of the auto-generated building blocks provided by CDK8s, this
...
+- Non stable resources are suffixed with their semantic version. For example, the `io.k8s.api.networking.v1beta1.Ingress` maps to the `IngressV1Beta1` *construct*.
```
Comment (eladb) [inline docs/plus/index.md:16]:
```suggestion
- Non stable resources are suffixed with their API version. For example, the `io.k8s.api.networking.v1beta1.Ingress` maps to the `IngressV1Beta1` *construct*.
```

**Review by eladb — APPROVED** [review]

**Pull request merged**